File: Rock Paper Scissors Game/Rock_Paper_Scissors_Game.c
#include<stdio.h>
#include<stdlib.h>
#include<time.h>
void playRound()
{
    int player,comp=(rand()%3)+1;
    while(1)
    {
        printf("Choose 1.rock 2.paper 3.scissors  ");
        scanf("%d",&player);
        if (player>=1 && player<=3)
        {
            break;
        }
        else
        {
            printf("Invalid input! Please enter 1, 2, or 3 only.\n");
        }
    }
    switch(player)
    {
        case 1:
            switch(comp)
            {
                case 1:
                printf("It's a draw!\n");
                break;
                case 2:
                printf("Sorry, you lost!\n");
                break;
                case 3:
                printf("Congratulations! You have won.\n");
                break;
            }
            break;
        case 2:
            switch(comp)
            {
                case 1:
                printf("Congratulations! You have won.\n");
                break;
                case 2:
                printf("It's a draw!\n");
                break;
                case 3:
                printf("Sorry, you lost!\n");
                break;
            }
            break;
        case 3:
            switch(comp)
            {
                case 1:
                printf("Sorry, you lost!\n");
                break;
                case 2:
                printf("Congratulations! You have won.\n");
                break;
                case 3:
            printf("It's a draw!\n");
                break;
            }
            break;
    }
}
int main()
{
    char choice;
    srand(time(0));
    while(1)
    {
        playRound();
        while(1)
        {
            printf("\nDo you want to play again? (y/n): ");
            scanf(" %c",&choice);
            if (choice=='y' || choice=='Y' || choice=='n' || choice=='N')
            {
                break;
            }
            else
            {
                printf("Invalid input! Please enter 1, 2, or 3 only.\n");
            }
        }
        if (choice=='y' || choice=='Y')
        {
            continue;
        }
        else
        {
            break; 
        }
    }
    return 0;
}

File: Rock Paper Scissors Game/Rock_Paper_Scissors_Game.py
import random
def playRound():
    while True:
        try:
            player=int(input("Choose 1.rock 2.paper 3.scissors: "))
            if player<1 or player>3:
                print("Invalid input! Please enter 1, 2, or 3 only.\n")
                continue
            break
        except ValueError:
            print("Invalid input! Please enter 1, 2, or 3 only.\n")
    comp=random.randint(1,3)
    match player:
        case 1:
            match comp:
                case 1:
                    print("It's a draw!")
                case 2:
                    print("Sorry, you lost!")
                case 3:
                    print("Congratulations! You have won.")
        case 2:
            match comp:
                case 1:
                       print("Congratulations! You have won.")
                case 2:
                    print("It's a draw!")
                case 3:
                    print("Sorry, you lost!") 
        case 3:
            match comp:
                case 1:
                       print("Sorry, you lost!")
                case 2:
                    print("Congratulations! You have won.")
                case 3:
                    print("It's a draw!")
playRound()
while True:
    try:
        choice=input("\nDo you want to play again? (y/n): ")
        print()
    except:
        print("Invalid input! Please enter 1, 2, or 3 only.\n")
    if choice=='y' or choice == 'Y':
        playRound()
    else:
        print("Thanks for playing!")
        exit()